lambda/translcore.ml

```diff
@@ -237,7 +237,7 @@ let rec push_defaults loc bindings use_lhs cases partial warnings =
                  desc, Id_value)},
              cases, partial) }
       in
-      [{c_lhs = {pat with pat_desc = Tpat_var (param, mknoloc name)};
+      [{c_lhs = {pat with pat_desc = Tpat_var (param, mknoloc name, Value_mode.global)};
         c_guard = None; c_rhs= wrap_bindings bindings exp}]
```

Comment:
I was unhappy with this so I've pushed another commit to plumb arg_mode through typedtree, so that we can choose a correct mode here. (This will go away with n-ary lambdas)